fluid/DeepASR/data_utils/load_data.py

```diff
@@ -0,0 +1,183 @@
+#by zhxfl 2018.01.24
+""" @package docstring
+load speech data from disk
+"""
+
+import random
+import Queue
+import numpy
+import struct
+import data_utils.trans_mean_variance_norm as trans_mean_variance_norm
+import data_utils.trans_add_delta as trans_add_delta
+
+g_lblock = []
+g_que_sample = Queue.Queue()
+g_nframe_dim = 120 * 11
+g_nstart_block_idx = 0
+g_nload_block_num = 1
+g_ndrop_frame_len = 256
+
+class OneBlock(object):
+    """ Documentation for a class.
+        struct for one block :
+        contain label, label desc, feature, feature_desc
+    """
+    def __init__(self):
+        """The constructor."""
+        self.label = ""
+        self.label_desc = ""
+        self.feature = ""
+        self.feature_desc = ""
+
+def set_trans(ltrans):
+    global g_ltrans
+    g_ltrans = ltrans
+
+def load_list(sfeature_lst, slabel_lst):
+    """ load list """
+    global g_lblock 
+
+    lFeature = open(sfeature_lst).readlines()
+    lLabel = open(slabel_lst).readlines()
+    assert len(lLabel) == len(lFeature)
+    for i in range(0, len(lFeature), 2):
+        one_block = OneBlock()
+
+        one_block.label = lLabel[i]
+        one_block.label_desc = lLabel[i + 1]
+        one_block.feature = lFeature[i]
+        one_block.feature_desc = lFeature[i + 1]
+        g_lblock.append(one_block)
+    
+    random.shuffle(g_lblock)
+
+def load_one_block(lsample, id):
+    """read one block"""
+    global g_lblock
+    if id >= len(g_lblock):
+        return
+
+    slabel_path = g_lblock[id].label.replace("\n", "")
+    slabel_desc_path = g_lblock[id].label_desc.replace("\n", "")
+    sfeature_path = g_lblock[id].feature.replace("\n", "")
+    sfeature_desc_path = g_lblock[id].feature_desc.replace("\n", "")
+
+    llabel_line = open(slabel_desc_path).readlines()
+    lfeature_line = open(sfeature_desc_path).readlines()
+    
+    file_lable_bin = open(slabel_path, "r")
+    file_feature_bin = open(sfeature_path, "r")
+    
+    sample_num = int(llabel_line[0].split()[1])
+    assert sample_num == int(lfeature_line[0].split()[1])
+
+    llabel_line = llabel_line[1:]
+    lfeature_line = lfeature_line[1:]
+
+    for i in range(sample_num):
+        # read label 
+        llabel_split = llabel_line[i].split()
+        nlabel_start = int(llabel_split[2])
+        nlabel_size = int(llabel_split[3])
+        nlabel_frame_num = int(llabel_split[4])
+
+        file_lable_bin.seek(nlabel_start, 0)
+        label_bytes = file_lable_bin.read(nlabel_size)
+        assert nlabel_frame_num * 4 == len(label_bytes)
+        label_array = struct.unpack('I' * nlabel_frame_num, label_bytes)
+        label_data = numpy.array(label_array, dtype=int)
```

Comment:
We can unify the `dtype` for label data, in Fluid `int64` is commonly used.
dtype='int64'